docs/development/authentication.md

```diff
@@ -0,0 +1,137 @@
+# Understanding authentication at runtime
+
+The following document provides an introduction around the different authentication methods that can take place during an image build when using the Build operator.
+
+- [Overview](#overview)
+- [Authentication for Git](#authentication-for-git)
+  - [Basic authentication](#basic-authentication)
+  - [SSH authentication](#ssh-authentication)
+  - [Usage of git secret](#usage-of-git-secret)
+- [Authentication to container registries](#authentication-to-container-registries)
+  - [Docker Hub](#docker-hub)
+  - [Usage of registry secret](#usage-of-registry-secret)
+- [References](#references)
+
+## Overview
+
+There are two places where users might need to define authentication when building images. Authentication to a container registry is the most common one, but also users might have the need
+to define authentications for pulling source-code from Git.
+
+## Authentication for Git
+
+There are two ways for authenticating into Git(_applies to both Gitlab or Github_). Also, see more information in the official Tekton [docs](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#configuring-ssh-auth-authentication-for-git).
```

Comment:
```suggestion
There are two ways for authenticating into Git (_applies to both GitLab or GitHub_). Also, see more information in the official Tekton [docs](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#configuring-ssh-auth-authentication-for-git).
```

Reply:
@HeavyWombat more than happy seeing you reviewing code. Done